dynamometer-infra/src/main/java/com/linkedin/dynamometer/AMOptions.java

```diff
@@ -59,7 +63,7 @@
   AMOptions(int datanodeMemoryMB, int datanodeVirtualCores, String datanodeArgs,
       String datanodeNodeLabelExpression, int datanodesPerCluster, String datanodeLaunchDelay, int namenodeMemoryMB,
       int namenodeVirtualCores, String namenodeArgs, String namenodeNodeLabelExpression, int namenodeMetricsPeriod,
-      Map<String, String> shellEnv) {
+      String namenodeNameDir, String namenodeEditsDir, Map<String, String> shellEnv) {
```

Comment:
this constructor is growing huge and a lot have default values, shall we construct this through a builder?

Reply:
My initial thought was to agree with you, but actually, I think it makes sense for it to stay as-is. Since this is just an internal class for encapsulating the result of a number of command-line options, you wouldn't normally call this constructor directly, and in fact the only current usage is within this class (`initFromParser()`). It does not seem worth creating a builder given this setup.